### ISSUE TITLE
clang-tidy readability-identifier-naming reports false-positive for template methods

### ISSUE THREAD

**Issue by lazolv** [llvm/llvm-project#189755]
main.cpp:
```cpp
class foo {
public:
    template <typename t>
    void doStuff() {}
};

int main() {
    return 0;
}
```
.clang-tidy:
```yml
Checks: >
  readability-identifier-naming,
# I set all options I know that could possibly have any affect on this
CheckOptions:
  - { key: readability-identifier-naming.DefaultCase,                    value: lower_case }
  - { key: readability-identifier-naming.FunctionCase,                   value: camelBack }
  - { key: readability-identifier-naming.MemberCase,                     value: camelBack }
  - { key: readability-identifier-naming.ClassMemberCase,                value: camelBack }
  - { key: readability-identifier-naming.MethodCase,                     value: camelBack }
  - { key: readability-identifier-naming.ClassMethodCase,                value: camelBack }
```
Running `clang-tidy main.cpp` outputs only 1 warning instead of 0:
```
(...)/main.cpp:4:10: warning: invalid case style for identifier 'doStuff' [readability-identifier-naming]
    4 |     void doStuff() {}
      |          ^~~~~~~
      |          do_stuff
```
Meanwhile, turning `doStuff()` into a regular method doesn't produce any warnings, as expected

I also found out that this behavior first appeared in [LLVM 22.1.0-rc1](https://github.com/llvm/llvm-project/releases/tag/llvmorg-22.1.0-rc1), clang-tidy from [LLVM 21.1.8](https://github.com/llvm/llvm-project/releases/tag/llvmorg-21.1.8) outputs no warnings
